tests/shell/mod.rs

```diff
@@ -140,6 +140,36 @@ fn nu_lib_dirs_relative_repl() {
     })
 }
 
+// TODO: add absolute path tests after we expand const capabilities (see #8310)
+#[test]
+fn const_nu_lib_dirs_relative() {
+    Playground::setup("const_nu_lib_dirs_relative", |dirs, sandbox| {
+        sandbox
+            .mkdir("scripts")
+            .with_files(vec![FileWithContentToBeTrimmed(
+                "scripts/foo.nu",
+                r#"
+                    let-env FOO = "foo"
+                "#,
+            )])
+            .with_files(vec![FileWithContentToBeTrimmed(
+                "main.nu",
+                r#"
+                    const NU_LIB_DIRS = [ 'scripts' ]
+                    source-env foo.nu
+                    $env.FOO
+                "#,
+            )]);
+
+        let inp_lines = &[r#"source main.nu"#];
+
+        let actual_repl = nu!(cwd: dirs.test(), nu_repl_code(inp_lines));
```

Comment:
This could be tested using normal execution, REPL simulator should not be necessary.

Reply:
Fixed